### PR TITLE
[SYCL] [libdevice] Add vector overloads of ConvertBFloat16ToFINTEL and ConvertFToBFloat16INTEL

### PR DESCRIPTION
**Pull request by uditagarwal97**
This PR adds vector overloads of `ConvertBFloat16ToFINTEL` and `ConvertFToBFloat16INTEL` to libdevice (SPEC: https://spec.oneapi.io/level-zero/latest/core/SPIRV.html#bfloat16-conversions) and a wrapper around it (`BF16VecToFloatVec` and `FloatVecToBF16Vec`) in `ext::oneapi::detail`.

These overloads are intended to optimize BFloat16 `marray`, `vec` operations, for which we currently do element-by-element `bfloat16 -> float -> bfloat16` conversions.

### CODE REVIEW

**Comment by uditagarwal97** [comment]
@intel/dpcpp-tools-reviewers friendly ping!

**Review by asudarsa — APPROVED** [review]
sycl-post-link changes are trivial and look good. Thanks.

File: sycl/test-e2e/BFloat16/bfloat16_conversions.cpp [hunk 1]
```diff
@@ -55,6 +58,82 @@ template <typename T> int test_host() {
...
+       for (int i = 0; i < 3; i++) {
+         if (FloatArray[i] != (float)BF16Array[i]) {
+           ERR[0] = 1;
+         }
+       }
+     });
+   }).wait();
+
+  if (err)
+    std::cout << "failed\n";
```
Comment (cperkinsintel) [inline sycl/test-e2e/BFloat16/bfloat16_conversions.cpp:130]: if there is a future change in the source that causes this test to fail someday, it'll be more work for whoever catches that ticket.  Make the failure output more explicit.  At minimum output "failed on host" and "failed on device". And if you know of anything else that might be useful to some unlucky co-worker, maybe include that too.

**Review by cperkinsintel — APPROVED** [review]
Lime Gin Tastes Metallic.  

just one request for better test output

File: sycl/test-e2e/BFloat16/bfloat16_conversions.cpp (hunk 1 above)
Comment (uditagarwal97) [reply sycl/test-e2e/BFloat16/bfloat16_conversions.cpp:130]: Sure. I'll improve the test output.

Comment (uditagarwal97) [reply sycl/test-e2e/BFloat16/bfloat16_conversions.cpp:130]: Actually, after looking at the test again, the test will currently output something like:
```
    bfloat16[4] -> float[4] conversion on device...  passed/failed
```
or
```
float[4] -> bfloat16[4] -> float[4] conversion on host.... passed/failed
```
for device and host respectively.  There's a `std::cout` before `queue.submit()` which will output the type of conversion being performed along where the conversion takes place (`host`/`device`).

Comment (bader) [inline sycl/test-e2e/BFloat16/bfloat16_conversions.cpp:130]: Shouldn't this go to `std::cerr` instead of `std::cout`?

**Pull request merged**